packages/react-tabs/src/components/Tab/Tab.test.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import { Tab } from './Tab';
+import { isConformant } from '../../common/isConformant';
+import { TabListContext } from '../TabList/TabListContext';
+import { TabListContextValue } from '../TabList/TabList.types';
+import { CalendarMonth24Regular } from '@fluentui/react-icons';
+
+describe('Tab', () => {
+  isConformant({
+    Component: Tab,
+    displayName: 'Tab',
+  });
+
+  const defaultContext: TabListContextValue = {
+    appearance: 'transparent',
+    size: 'medium',
+    vertical: false,
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    onSelect: () => {},
+  };
+
+  it.each([
+    ['default', { ...defaultContext }],
+    ['subtle appearance', { ...defaultContext, appearance: 'subtle' }],
+    ['vertical', { ...defaultContext, vertical: true }],
+    ['small size', { ...defaultContext, size: 'small' }],
+    ['small size and vertical', { ...defaultContext, size: 'small', vertical: true }],
+    ['selected', { ...defaultContext, selectedValue: '1' }],
+    ['not selected', { ...defaultContext, selectedValue: '2' }],
+  ])('renders %s correctly', (_testName, tabList) => {
+    const contextValues = {
+      tabList: tabList as TabListContextValue,
+    };
+
+    const result = render(
+      <TabListContext.Provider value={contextValues.tabList}>
+        <Tab value="1">Default Tab</Tab>
+      </TabListContext.Provider>,
+    );
+
+    expect(result.container).toMatchSnapshot();
+  });
+
+  it('selected when clicked', () => {
+    const onSelect = jest.fn();
+
+    const contextValues = {
+      tabList: { ...defaultContext, onSelect },
+    };
+
+    const result = render(
+      <TabListContext.Provider value={contextValues.tabList}>
+        <Tab value="1">Default Tab</Tab>
+      </TabListContext.Provider>,
+    );
+
+    fireEvent.click(result.getByRole('tab'));
+    expect(onSelect).toHaveBeenCalledWith(expect.anything(), { value: '1' });
+  });
```

Comment:
(nit) I wonder if this would be a better test for `TabList`, so you're not having to manually create a `TabListContext` for the test.

Does `Tab` have an `onSelect` event of its own to test here, that's not on the context? (Also, if not, it probably should?)

Reply:
Tab doesn't have an onSelect event of its own.  I didn't add it because I was following MenuItem, but I realize that Menus use onClick which does happen on MenuItem because it is an intrinsic event.  I'll add it to Tab.  Great catch!